synapse/api/errors.py

```diff
@@ -600,8 +606,12 @@ def cs_error(msg: str, code: str = Codes.UNKNOWN, **kwargs: Any) -> "JsonDict":
 
 
 class FederationError(RuntimeError):
-    """This class is used to inform remote homeservers about erroneous
-    PDUs they sent us.
+    """
+    This class is used signal about erroneous PDUs that we pulled from a remote homeserver.
+
+    It's also used inform remote homeservers that the PDU they sent us (via `/send`) is
+    erroneous. These error responses and status codes are not specced so the remote
+    homeserver probably won't do anything about it.
```

Comment:
AFAICT, these error responses are not spec'd, https://spec.matrix.org/v1.1/server-server-api/#put_matrixfederationv1sendtxnid

And `_handle_received_pdu` says that:

https://github.com/matrix-org/synapse/blob/40bb37eb27e1841754a297ac1277748de7f6c1cb/synapse/federation/federation_server.py#L1070-L1073

Reply:
Do you have an example? The spec makes it look like we're always supposed to return 200 OK with a `PDU Processing Results` object. Does this exception mean we don't return 200 in that situation? Or maybe it's used to generate an error object for that event in the "PDU Processing Results"?

Reply:
> Or maybe it's used to generate an error object for that event in the "PDU Processing Results"?

Ahh, I see what you're getting at 💡. I think it does exactly what you described!

https://github.com/matrix-org/synapse/blob/1c777ef1e87d7be39a2b8f6fb119fa4b51e2be4c/synapse/federation/federation_server.py#L487-L491

Sorry, I am not familiar with federation push `/send` and didn't read the spec close enough 🙇

I've updated the error docstring and added some references to `"PDU Processing Result"` in the code where we construct them to cross-reference.

Reply:
And see also here where the /send response looks like it's being built:

https://github.com/matrix-org/synapse/blob/1c777ef1e87d7be39a2b8f6fb119fa4b51e2be4c/synapse/federation/federation_server.py#L480-L481